crates/nu-command/src/filesystem/ls.rs

```diff
@@ -175,22 +176,33 @@ impl Command for Ls {
             ));
         }
 
-        let mut hidden_dirs = vec![];
+        let hidden_dirs = Arc::new(Mutex::new(Vec::new()));
 
         Ok(paths_peek
+            .into_iter()
+            .par_bridge()
```

Comment:
I love how easy it is to try stuff like this out.

I think `paths_peek` will reorder the files in directories, if I'm reading the original correctly. This will likely annoy folks, who will be used to files showing up in a predictable order so they can then operate on the files like (ls | get 3) and so on.

I think if we want to do this, we could either a) make a flag on `ls` that runs `ls` in parallel. or b) only runs the separate directories in parallel but leaves their entries in the same order as before.

Reply:
the order is clearly non-deterministic now, as you can see by all the tests i had to change. i'd vote against the --parallel flag, i think it should be enabled by default so we always get the speed we need. An option c is to always sort the data in a particular way. i'd vote for sorting by type then name.